### PR TITLE
Add push functionality to CDS Handler

### PR DESCRIPTION
**Pull request by stelabouras**
Adds functionality to push base translations to CDS by introducing the
`pushTranslations()` method in `TxNative`, `NativeCore` and `CDSHandler`
classes.

Alongside this method, the `TxTranslationUnit` class is also introduced
which offers a way for both Swift and Objective-C projects to construct
the data structure that will hold all the necessary information for the
strings to be pushed. The `TxTranslationUnit` also offers an internal
method that transforms the object to the `Codable` `SourceString` struct
that can be used when constructing the data to be pushed.

This commit also moves the `occurencies` property from `SourceString`
struct to `SourceString` meta, where it should have been originally
placed.

The `pushTranslations()` method of `CDSHandler` serializes the passed
array of `TxTranslationUnit` objects using an internal `PushData` struct
and the resulting `Data` object is passed to the `URLRequest` that is
used to push the strings to CDS. An extra optional `purge` parameter can
also be used to replace the entire resource content, if needed.

When the push logic completes, a completion handler informs the caller
asynchronously on whether the request was successful or not.

Finally this commit removes a couple of unused `;` characters.

### CODE REVIEW

**Review by dbendilas — CHANGES_REQUESTED** [review]
Looks great. I have a question about `TxTranslationUnit`, the rest of my comments are trivial.

File: Sources/TransifexNative/CDSHandler.swift [hunk 1]
```diff
@@ -207,23 +223,83 @@ class CDSHandler {
...
-                }
-            } catch {
-                print("Error encoding source string \(string): \(error.localizedDescription)")
+    /// - Parameters:
+    ///   - translations: A list of `TxTranslationUnit` objects
+    ///   - purge: Whether the request will replace the entire resource content (true) or not (false)
+    ///   Defaults to false
+    ///   - completionHandler: a callback function to call when the operation is complete
+    public func pushTranslations(_ translations: [TxTranslationUnit],
+                                 _ purge: Bool = false,
```
[the remark this answers is not in the document]
Comment (stelabouras) [reply Sources/TransifexNative/CDSHandler.swift]: Done!

File: Sources/TransifexNative/SourceString.swift [hunk 2]
```diff
@@ -8,18 +8,73 @@
...
+/// Publicly exposed structure that holds all the information for the translation units that will be pushed to CDS.
+public class TxTranslationUnit: NSObject {
+    /// The key that was generated by the `generateKey()` method
+    let key: String
+    /// The source string
+    let sourceString: String
+    /// An optional developer comment
+    let developerComment: String?
+    /// A list of relative file paths where this string is located.
+    let occurencies: [String]
```
[the remark this answers is not in the document]
Comment (stelabouras) [reply Sources/TransifexNative/SourceString.swift]: Good catch, fixed!

File: Sources/TransifexNative/SourceString.swift [hunk 3]
```diff
@@ -8,18 +8,73 @@
 
 import Foundation
 
+/// Publicly exposed structure that holds all the information for the translation units that will be pushed to CDS.
+public class TxTranslationUnit: NSObject {
```
[the remark this answers is not in the document]
Comment (stelabouras) [reply Sources/TransifexNative/SourceString.swift:12]: `SourceString` is a Swift struct that conforms to the `Codable` protocol for JSON serialization / deserialization, so we can't use it on Objective-C projects.

I wanted to keep the functionality of that `Codable` struct internally, but offer a way for both Swift / Objective-C projects to pass the translation units to the SDK.

I am open to other suggestions though!

**Review by dbendilas — APPROVED** [review]

**Pull request merged**